app/models/publisher.rb

```diff
@@ -48,39 +48,28 @@ def save_catching_gds_api_errors(&block)
         if content_model.save
           block.call
 
-          PublicationResponse.new(success: true)
+          Response.new(success: true)
         else
-          PublicationResponse.new(success: false)
+          Response.new(success: false)
         end
       end
     rescue GdsApi::HTTPErrorResponse => e
-      PublicationResponse.new(success: false, errors: e.error_details['error']['message'])
+      Response.new(success: false, error: e.error_details['error']['message'])
     end
   end
 
-  class DiscardDraftResponse
+  class Response
     def initialize(opts = {})
       @success = opts.fetch(:success)
-      @errors = opts.fetch(:errors, [])
+      @error = opts.fetch(:error, nil)
```

Comment:
There's no need for an array here though? There's always only one error returned.